app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt

```diff
@@ -380,16 +402,13 @@ class SubscriptionFragment : BaseStateFragment<SubscriptionState>() {
     override fun handleResult(result: SubscriptionState) {
         super.handleResult(result)
 
-        val shouldUseGridLayout = shouldUseGridLayout(context)
+        val itemViewMode = getItemViewMode(requireContext())
         when (result) {
             is SubscriptionState.LoadedState -> {
                 result.subscriptions.forEach {
                     if (it is ChannelItem) {
                         it.gesturesListener = listenerChannelItem
-                        it.itemVersion = when {
-                            shouldUseGridLayout -> ChannelItem.ItemVersion.GRID
-                            else -> ChannelItem.ItemVersion.MINI
-                        }
+                        it.itemVersion = getItemVersion(itemViewMode)
```

Comment:
what about (and also removing the `getItemVersion`, maybe move its description to `shouldUseGridForSubscription`)
```suggestion
                        it.itemVersion = if shouldUseGridForSubscription()
                            ChannelItem.ItemVersion.GRID
                        else
                            ChannelItem.ItemVersion.MINI
```

Reply:
Done